documentation/sphinx/source/developer-guide.rst

```diff
@@ -756,6 +756,118 @@ If you only need to detect the *fact* of a change, and your response doesn't dep
 
 .. _developer-guide-peformance-considerations:
 
+
+Special keys
+============
+
+Keys starting with the bytes ``\xff\xff`` are called "special" keys, and they are materialized when read. :doc:`\\xff\\xff/status/json <mr-status>` is an example of a special key.
+As of api version 630, additional features have been exposed as special keys and are available to read as ranges instead of just individual keys. Additionally, the special keys are now organized into "modules".
+
+Modules
+-------
+
+A module is loosely defined as a key range where a user can expect similar behavior from reading any key in that range.
+By default, users will see a ``special_keys_no_module_found`` error if they read from a range not contained in a module.
+This is likely to be an error since the read would always return an empty set of keys and can't be interesting (maybe it was a typo?).
+Users will also (by default) see a ``special_keys_cross_module_read`` error if their read spans a module boundary.
+This is also potentially an error, since the user may be surprised by seeing the behavior of multiple modules in the same read.
```

Comment:
no need to explain the rationale, so that the flow is smoother.

Reply:
The rationale is important so the user can decide if they want to set the special_key_space_relaxed option

Reply:
The I prefer to rewording the sentence to `The error is to save the user from the surprise of seeing the behavior of multiple modules in the same read`